src/client.ts

```diff
@@ -556,6 +563,28 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
     return partialClientConfig?.endpoint ?? env_endpoint ?? endpoints.default;
   }
 
+  #getLogger(partialClientConfig?: ClientConfiguration): LogHandler {
+    if (
+      partialClientConfig &&
+      "logger" in partialClientConfig &&
+      partialClientConfig.logger === undefined
+    ) {
+      throw new TypeError(`ClientConfiguration option logger must be defined.`);
+    }
+    if (
+      typeof process != "undefined" &&
+      process &&
+      typeof process === "object" &&
+      process.env &&
+      typeof process.env === "object"
+    ) {
+      if (faunaDebugEnabled(process.env["FAUNA_DEBUG"])) {
+        return new ConsoleLogHandler(LogLevel.DEBUG);
+      }
+    }
+    return new ConsoleLogHandler(LogLevel.ERROR);
+  }
```

Comment:
Anything that the user provides in the `logger` option is ignored, other than checking that it is undefined. We should return the explicitly given object if provided and fall back on the env variables

```suggestion
  #getLogger(partialClientConfig?: ClientConfiguration): LogHandler {
    if (
      partialClientConfig &&
      "logger" in partialClientConfig &&
      partialClientConfig.logger === undefined
    ) {
      throw new TypeError(`ClientConfiguration option logger must be defined.`);
    }
    
    if (partialClientConfig?.logger) {
      return logger
    }
    
    let env_debug: string | undefined = undefined;
    if (
      typeof process !== "undefined" &&
      process &&
      typeof process === "object" &&
      process.env &&
      typeof process.env === "object"
    ) {
      env_debug = process.env["FAUNA_DEBUG"]
    }
    
    if (faunaDebugEnabled(env_debug)) {
      return new ConsoleLogHandler(LogLevel.DEBUG);
    }
    
    return new ConsoleLogHandler(LogLevel.ERROR);
  }
```

Reply:
I _think_ we also want to pass in the log level provided to the `ConsoleLogHandler`, rather than hardcoding DEBUG and ERROR.